### DISCUSSION TITLE
Issue while creating a docker image of Go with nuclei and python

### DISCUSSION THREAD

**Discussion by lijoabraham** [projectdiscovery/nuclei#3767] [Q&A]
### Nuclei version: 
2.9.5

### Current Behavior:
Getting the below error while creating a docker image of Go with nuclei and python.

<img width="601" alt="image" src="https://github.com/projectdiscovery/nuclei/assets/119271231/519756c1-6ffa-4a0f-9556-28a5c1037f99">

'Dockerfile'

```
# Base image
FROM golang:latest

# Install necessary dependencies
RUN apt-get update && apt-get install -y git

# Install Nuclei
RUN go install -v github.com/projectdiscovery/nuclei/v2/cmd/nuclei@latest

# Install Python
RUN apt-get install -y python3 python3-pip

# Copy your Python script into the container
COPY . .

# Set the working directory
WORKDIR /app

# Run your Python script
CMD ["python3", "/app/nuclei.py"]

```

and python code 'nuclei.py' is 

```
import subprocess

def run_nuclei(target, template):
    command = f'nuclei -target {target} -ud {template}'
    try:
        output = subprocess.check_output(command, shell=True, text=True, stderr=subprocess.STDOUT)
        return output
    except subprocess.CalledProcessError as e:
        # Handle any errors that occurred during execution
        print(f"Error executing Nuclei: {e.output}")
        return None

# Usage example
target_url = "https://google.com"
template_path = "/app/nuclei-templates"
result = run_nuclei(target_url, template_path)
if result:
    print(result)

```
### Expected Behavior:
docker image should build without giving any error

### Steps To Reproduce:
1. Create the above mentioned files.
2. run the command - `docker build -t golang-nuclei-python . '

**Comment by olearycrew** [comment]
@lijoabraham can you try to install curl first in the Docker container? We saw similar behavior here: https://github.com/projectdiscovery/nuclei/discussions/2875

**Comment by lijoabraham** [comment]
@olearycrew still same.
<img width="599" alt="image" src="https://github.com/projectdiscovery/nuclei/assets/119271231/9c3d48ea-074c-4e70-b0ec-18d2165132ef">

**Comment by ehsandeep** [comment]
Moving this to the discussion as this is not an issue in nuclei and more of a networking issue on the host machine.

**Comment by MetzinAround** [comment]
https://stackoverflow.com/questions/50768317/docker-pull-certificate-signed-by-unknown-authority

https://forums.docker.com/t/docker-private-registry-x509-certificate-signed-by-unknown-authority/21262

Found these issues online with Docker. Some insight on there might help.

**Comment by lijoabraham** [comment]
@MetzinAround I was able to got it working by using the below docker file commands.

```
# Base image
FROM projectdiscovery/nuclei:latest

WORKDIR /app
# Install Python
ENV PYTHONUNBUFFERED=1
RUN apk add --update --no-cache python3 && ln -sf python3 /usr/bin/python
RUN python3 -m ensurepip
RUN pip3 install --no-cache --upgrade pip setuptools

COPY requirements.txt .
RUN pip3 install --no-cache -r requirements.txt


# Copy your Python script into the container
COPY . /app

# Run your Python script
CMD ["bash"]
```

**Reply by MetzinAround** [reply]
Awesome! Thanks for the update.
